### DISCUSSION TITLE
Create a workplane perpendicular to a line

### DISCUSSION THREAD

**Discussion by Jopie01** [CadQuery/cadquery#1596] [Q&A] [answered]
I want to create a sweep along a line. So I create a line first as the path and then I want to create the section to be swept along the line. To do that, I have to calculate different angels and positions. Is it possible to add a workplane at the end of the path and perpendicular to the that path?

```python
path = (cq.Workplane("YZ")
    .transformed(offset=(-200, -30, 340))
    .lineTo(3896, 2310)
)

p = (path.edges().workplane()
    .rect(40, 40)
)

show_object([path, p])
```

`p` in this case is at the end of the path, but not perpendicular.

**Comment by lorenzncode** [comment, marked as answer]
Here is an example (free function API in master branch):
Edit Jan 2026: Change sweep call to correct usage with the released version ( `cap=True`).
```py
import cadquery as cq
from cadquery.occ_impl.shapes import *

pts = [(10, 10, 0), (20, 10, 10), (30, 10, -10), (30, 0, 0)]

path = spline(pts)
# create plane at start of path with normal tangent to path
plane = cq.Plane(origin=path.startPoint(), normal=path.tangentAt(0))
profile = cq.Workplane(plane).polygon(3, 1).val()
sweep = sweep(profile, path, cap=True)
```

In your case with the path Edge in Workplane:

```py
import cadquery as cq

path = (cq.Workplane("YZ")
        .transformed(offset=(-200, -30, 340))
        .lineTo(3896, 2310)
        )

plane = cq.Plane(origin=path.val().endPoint(), normal=path.val().tangentAt(1))
p = cq.Workplane(plane).rect(40, 40)
res = p.sweep(path)
```

**Reply by Jopie01** [reply]
Thanks a lot! It works perfectly.
